### DISCUSSION TITLE
More convenient multiline prompt

### DISCUSSION THREAD

**Discussion by johnd0e** [darrenburns/elia#36] [Ideas]
At the moment the prompt is unconditionally multiline, to submit it user must press <kbd>Ctrl</kbd>+<kbd>J</kbd>.

To make it more convenient I propose the following enhancements:
- single-line mode
  - The prompt is considered single-line by default
  - <kbd>Enter</kbd> to submit
  -  <kbd>Ctrl</kbd>+<kbd>Enter</kbd> to add new line (or <kbd>Shift</kbd>+<kbd>Enter</kbd>)
- (optional) "smart" <kbd>Enter</kbd> in multiline mode
  - Single <kbd>Enter</kbd> to add new line
  - Double <kbd>Enter</kbd> to submit
  - (no need) ~<kbd>Ctrl</kbd>+<kbd>Enter</kbd> (or <kbd>Shift</kbd>+<kbd>Enter</kbd>) to force new line unconditionally~

**Comment by darrenburns** [comment]
Unfortunately Ctrl+Enter / Shift+Enter isn't supported in terminals - that's why `Ctrl+J` is chosen. It's an awkward problem.

At the terminal emulator level you could probably map <kbd>Ctrl</kbd>+<kbd>Enter</kbd> to <kbd>Ctrl</kbd>+<kbd>J</kbd>. I might suggest this in the README.

I quite like the idea of the "smart" enter though - I wonder how that might play out in practice. It'd likely need to be a config option as it could be annoying for some folk.

**Reply by johnd0e** [reply]
> Unfortunately Ctrl+Enter / Shift+Enter isn't supported in terminals - that's why Ctrl+J is chosen.

For the sake of simplicity: let's stay <kbd>Ctrl</kbd>+<kbd>J</kbd> for all purposes.
- multiline: for submit (current behavior)
- singleline: for new line (in other words: <kbd>Ctrl</kbd>+<kbd>J</kbd> and <kbd>Enter</kbd> swap their actions)

**Reply by johnd0e** [reply]
> At the terminal emulator level you could probably map Ctrl+Enter to Ctrl+J

For me Ctrl+Enter works out-of-the-box now
